tests/bluetooth/tester/src/audio/btp_bap_audio_stream.h

```diff
@@ -2,23 +2,82 @@
 
 /*
  * Copyright (c) 2023 Codecoup
+ * Copyright (c) 2024 Nordic Semiconductor ASA
  *
  * SPDX-License-Identifier: Apache-2.0
  */
 
+#ifndef BTP_BAP_AUDIO_STREAM_H
+#define BTP_BAP_AUDIO_STREAM_H
+
+#include <stdint.h>
+
+#include <zephyr/bluetooth/conn.h>
+#include <zephyr/bluetooth/audio/bap.h>
 #include <zephyr/bluetooth/audio/cap.h>
+#include <zephyr/sys/util_macro.h>
+#include <zephyr/types.h>
 
 struct btp_bap_audio_stream {
 	struct bt_cap_stream cap_stream;
-	atomic_t seq_num;
-	uint16_t last_req_seq_num;
-	uint16_t last_sent_seq_num;
-	struct k_work_delayable audio_clock_work;
-	struct k_work_delayable audio_send_work;
 };
 
-int btp_bap_audio_stream_init_send_worker(void);
-void btp_bap_audio_stream_started(struct btp_bap_audio_stream *stream);
-void btp_bap_audio_stream_stopped(struct btp_bap_audio_stream *a_stream);
-uint8_t btp_bap_audio_stream_send(const void *cmd, uint16_t cmd_len, void *rsp, uint16_t *rsp_len);
-uint8_t btp_bap_audio_stream_send_data(const uint8_t *data, uint8_t data_len);
+/**
+ * @brief Initialize TX
+ *
+ * This will initialize TX if not already initialized. This creates and starts a thread that
+ * will attempt to send data on all streams registered with btp_bap_audio_stream_tx_register().
+ */
+void btp_bap_audio_stream_tx_init(void);
+
+/**
+ * @brief Register a stream for TX
+ *
+ * This will add it to the list of streams the TX thread will attempt to send on.
+ *
+ * @param bap_stream The stream to register
+ *
+ * @retval 0 on success
+ * @retval -EINVAL @p bap_stream is NULL
+ * @retval -EINVAL @p bap_stream is not configured for TX
+ * @retval -EINVAL @p bap_stream.codec_cfg contains invalid values
+ * @retval -ENOMEM if not more streams can be registered
+ */
+int btp_bap_audio_stream_tx_register(struct btp_bap_audio_stream *stream);
+
+/**
+ * @brief Unregister a stream for TX
+ *
+ * This will remove it to the list of streams the TX thread will attempt to send on.
+ *
+ * @param bap_stream The stream to unregister
+ *
+ * @retval 0 on success
+ * @retval -EINVAL @p bap_stream is NULL
+ * @retval -EINVAL @p bap_stream is not configured for TX
+ * @retval -EALREADY @p bap_stream is currently not registered
+ */
+int btp_bap_audio_stream_tx_unregister(struct btp_bap_audio_stream *stream);
+
+/**
+ * @brief Test if the provided stream has been configured for TX
+ *
+ * @param bap_stream The stream to test for TX support
+ *
+ * @retval true if it has been configured for TX, and false if not
+ */
+bool btp_bap_audio_stream_can_send(struct btp_bap_audio_stream *stream);
+
+/**
+ * @brief Callback to indicate a TX complete
+ *
+ * @param stream The stream that completed TX
+ */
+void btp_bap_audio_stream_sent_cb(struct bt_bap_stream *stream);
+
+/* Generate 510 octets of mock data going 0x00, 0x01, ..., 0xff, 0x00, 0x01, ..., 0xff, etc */
```

Comment:
Looks specific. I guess this is due to 255*2. But what happens if there are three streams?

Reply:
This is not a ringbuffers, it's just a static array. The number of streams does not affect this. I'll add additional reasoning why it's specifically 510 :)